suitest.js

```diff
@@ -0,0 +1,209 @@
+		// instance dependencies
+		this.authContext = new AuthContext();
+		this.appContext = new Context();
+		this.pairedDeviceContext = new Context();
+		this.testContext = new Context();
+		this.configuration = configFactory();
+		this.config = this.configuration.config;
+		this.configuration.configurableFields.map(fieldName => {
+			this[`set${fieldName[0].toUpperCase()}${fieldName.slice(1)}`] = (val) => this.configuration.override({[fieldName]: val});
+		});
+
+		// creating methods based on instance dependencies
+		this.logger = createLogger(this.config, this.pairedDeviceContext);
+		this.unusedExpressionWatchers = unusedExpressionWatchersFactory(this);
+		this.webSockets = webSocketsFactory(this);
+		setUpRaven(this.config, this.authContext);
+
+		this.openSession = (...args) => openSession(this, ...args);
+		this.pairDevice = (...args) => pairDevice(this, ...args);
+		this.setAppConfig = (...args) => setAppConfig(this, ...args);
+		this.closeSession = (...args) => closeSession(this, ...args);
+		this.startTestPack = (...args) => startTestPack(this, ...args);
+		this.startTest = (...args) => startTest(this, ...args);
+		this.endTest = (...args) => endTest(this, ...args);
+		this.releaseDevice = (...args) => releaseDevice(this, ...args);
+		this.interactive = interactive(this);
+
+		this.openApp = openAppFactory(this).openApp;
+		this.openUrl = openUrlFactory(this).openUrl;
+		this.application = applicationFactory(this).application;
+		this.clearAppData = clearAppDataFactory(this).clearAppData;
+		this.location = locationFactory(this).location;
+		this.cookie = cookieFactory(this).cookie;
+		this.sleep = sleepFactory(this).sleep;
+		this.press = pressButtonFactory(this).pressButton;
+		this.position = positionFactory(this).position;
+		this.window = windowFactory(this).window;
+		this.executeCommand = executeCommandFactory(this).executeCommand;
+		// this.executeBrightScript = executeBrightScriptFactory(this).executeBrightScript;
+		this.jsExpression = jsExpressionFactory(this).jsExpression;
+		// this.brightScriptExpression = brightScriptExpressionFactory(this).brightScriptExpression;
+		this.element = elementFactory(this).element;
+		this.networkRequest = networkRequestFactory(this).networkRequest;
+		this.video = videoFactory(this).video;
+		this.psVideo = playstationVideoFactory(this).playstationVideo;
+		this.pollUrl = pollUrlFactory(this).pollUrl;
+
+		this.PROP = ELEMENT_PROP;
+		this.COMP = PROP_COMPARATOR;
+		this.VALUE = VALUE;
+		this.VIDEO_STATE = VIDEO_STATE;
+		this.VISIBILITY_STATE = VISIBILITY_STATE;
+		this.CONTENT_MODE = CONTENT_MODE;
+		this.ELEMENT_STATE = ELEMENT_STATE;
+		this.TEXT_ALIGNMENT = TEXT_ALIGNMENT;
+		this.IMAGE_LOAD_STATE = IMAGE_LOAD_STATE;
+		this.BORDER_STYLE = BORDER_STYLE;
+		this.VRC = VRC;
+		this.KEY = KEY;
+		this.NETWORK_PROP = NETWORK_PROP;
+		this.NETWORK_METHOD = NETWORK_METHOD;
+		this.HAD_NO_ERROR = HAD_NO_ERROR;
+
+		this.assert = {
+			application: applicationFactory(this).applicationAssert,
```

Comment:
Not happy that factories are called twice just to get a single member from them each time. Should be called once per instance.